game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroupTest.java

```diff
@@ -0,0 +1,187 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.Comparator;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FiringGroupTest {
+
+  @Mock GameData gameData;
+  @Mock GamePlayer player;
+
+  @Test
+  public void onlyOneGroupIfNoSuicideOnHit() {
+    final List<Unit> units =
+        List.of(
+            givenUnitWithSuicideOnHit("type1", false), givenUnitWithSuicideOnHit("type2", false));
```

Comment:
Nit, the method name " givenUnitWithSuicideOnHit("type1", false)" is a bit akward with those parameters. I would have expected it to always return a unit with suicide on hit. Boolean parameters are generally evil, so one thing we can do is to create two methods for both cases and that would help with the naming problem. Then it would be:
```
givenUnitWithTypeNameAndSuicideOnHit("type1")
givenUnitWithTypeNameWithoutSuicideOnHit("type2")
```

Though, TBH we would do well if we could create a value object out of this and avoid mocks. That makes me think that the API we are testing should accept the property values rather than the actual unit type object.

If we keep the boolean parameter, then perhaps a more descriptive method name would be:
`givenUnitWithTypeAndHasSuicideOnHit(...)`

Reply:
The `givenUnit()` methods aren't returning mocks.  They are returning an actual `Unit` and/or `UnitType` object.  I've changed them to just return a default `UnitType` and/or `Unit` and then set the `isSuicideOnHit` in the test method itself.